nautobot/dcim/views.py

```diff
@@ -1668,66 +1670,60 @@ def bulk_rename(self, request, *args, **kwargs):
+    bulk_update_form_class = forms.PlatformBulkEditForm
+    filterset_class = filters.PlatformFilterSet
+    filterset_form_class = forms.PlatformFilterForm
+    form_class = forms.PlatformForm
+    serializer_class = serializers.PlatformSerializer
+    table_class = tables.PlatformTable
     queryset = Platform.objects.all()
 
-    def get_extra_context(self, request, instance):
-        # Devices
-        devices = (
-            Device.objects.restrict(request.user, "view")
-            .filter(platform=instance)
-            .select_related("status", "location", "tenant", "rack", "device_type", "role")
-        )
-
-        device_table = tables.DeviceTable(devices)
-
-        paginate = {
-            "paginator_class": EnhancedPaginator,
-            "per_page": get_paginate_count(request),
-        }
-        RequestConfig(request, paginate).configure(device_table)
-
-        return {
-            "device_table": device_table,
-            "network_driver_tool_names": get_network_driver_mapping_tool_names(),
-            **super().get_extra_context(request, instance),
-        }
-
-
-class PlatformEditView(generic.ObjectEditView):
-    queryset = Platform.objects.all()
-    model_form = forms.PlatformForm
-    template_name = "dcim/platform_edit.html"
+    object_detail_content = object_detail.ObjectDetailContent(
+        panels=(
+            object_detail.ObjectFieldsPanel(
+                weight=100,
+                section=SectionChoices.LEFT_HALF,
+                fields=["description", "manufacturer", "napalm_driver", "napalm_args", "network_driver"],
+            ),
+            object_detail.KeyValueTablePanel(
+                weight=100,
+                section=SectionChoices.RIGHT_HALF,
+                context_data_key="network_driver_tool_names",
+                label="Network Driver Mappings",
+            ),
+            object_detail.ObjectsTablePanel(
+                weight=100,
+                section=SectionChoices.FULL_WIDTH,
+                table_class=tables.DeviceTable,
+                table_filter="platform",
+                exclude_columns=["platform"],
+            ),
+        ),
+    )
 
     def get_extra_context(self, request, instance):
-        return {
-            "network_driver_names": sorted(get_all_network_driver_mappings().keys()),
-            **super().get_extra_context(request, instance),
-        }
-
-
-class PlatformDeleteView(generic.ObjectDeleteView):
-    queryset = Platform.objects.all()
-
+        if self.action == "retrieve":
+            devices = (
+                Device.objects.restrict(request.user, "view")
+                .filter(platform=instance)
+                .select_related("status", "location", "tenant", "rack", "device_type", "role")
+            )
 
-class PlatformBulkImportView(generic.BulkImportView):  # 3.0 TODO: remove, unused
-    queryset = Platform.objects.all()
-    table = tables.PlatformTable
+            device_table = tables.DeviceTable(devices)
```

Comment:
We can probably delete these lines since we already have `ObjectsTablePanel` on line 1695